prefect_dbt/cloud/jobs.py

```diff
@@ -630,3 +638,134 @@ def retry_dbt_cloud_job_run_subset_and_wait_for_completion_flow():
         poll_frequency_seconds=poll_frequency_seconds,
     )
     return run_data
+
+
+class DbtCloudJob(Block, ABC):
+    """
+    placeholder
+    """
+
+    _block_schema_capabilities = [
+        "trigger-job",
+        "wait-for-job-completion",
+        "fetch-job-results",
+    ]
+
+    dbt_cloud_credentials: DbtCloudCredentials
+
+    @property
+    def logger(self):
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger()
+
+    @sync_compatible
+    async def trigger(self, job_id, **trigger_job_run_options):
+        """
+        placeholder
+        """
+        triggered_run_data_future = await trigger_dbt_cloud_job_run.submit(
+            dbt_cloud_credentials=self.dbt_cloud_credentials,
+            job_id=job_id,
+            options=trigger_job_run_options,
+        )
+        run_id = (await triggered_run_data_future.result()).get("id")
+        self.logger.info("Triggered job {job_id!r} with run ID {run_id!r}")
+        return run_id
+
+    @sync_compatible
+    async def wait_for_completion(
+        self, run_id, max_wait_seconds, poll_frequency_seconds
```

Comment:
I think it makes sense to have `max_wait_seconds` and `poll_frequency_seconds` on the block as fields so that they are remotely configurable. `run_id` is an interesting one. Including it in the method signature will limit this interface to only jobs that have runs that can be identified by an ID, which is probably fine, but we might want to explore other ways of exposing this functionality.